src/PrpShop/PRP/Render/QPlasmaCubicRender.cpp

```diff
@@ -0,0 +1,412 @@
+    vec4 environmentColor = texture2D(envmap, v_texcoord);
+    vec4 labelColor = texture2D(labels, v_texcoord);
+    vec4 finalColor = mix(environmentColor, labelColor, labelColor.a * faceLabelBlendFactor);
+
+    gl_FragColor = finalColor;
+}
+)";
+
+struct VertexData {
+    QVector3D position;
+    QVector2D texCoord;
+};
+
+// Cube vertices and texture coordinates adapted
+// from Qt example code.
+constexpr VertexData vertices[] = {
+    // Vertex data for face 0
+    {{-1.0f, -1.0f,  1.0f}, {0.0f,      0.0f}}, //  v0
+    {{ 1.0f, -1.0f,  1.0f}, {1.f / 3.f, 0.0f}}, //  v1
+    {{-1.0f,  1.0f,  1.0f}, {0.0f,      0.5f}}, //  v2
+    {{ 1.0f,  1.0f,  1.0f}, {1.f / 3.f, 0.5f}}, //  v3
+
+    // Vertex data for face 1
+    {{ 1.0f, -1.0f,  1.0f}, {0.0f,      0.5f}}, //  v4
+    {{ 1.0f, -1.0f, -1.0f}, {1.f / 3.f, 0.5f}}, //  v5
+    {{ 1.0f,  1.0f,  1.0f}, {0.0f,      1.0f}}, //  v6
+    {{ 1.0f,  1.0f, -1.0f}, {1.f / 3.f, 1.0f}}, //  v7
+
+    // Vertex data for face 2
+    {{ 1.0f, -1.0f, -1.0f}, {2.f / 3.f, 0.5f}}, //  v8
+    {{-1.0f, -1.0f, -1.0f}, {1.0f,      0.5f}}, //  v9
+    {{ 1.0f,  1.0f, -1.0f}, {2.f / 3.f, 1.0f}}, // v10
+    {{-1.0f,  1.0f, -1.0f}, {1.0f,      1.0f}}, // v11
+
+    // Vertex data for face 3
+    {{-1.0f, -1.0f, -1.0f}, {2.f / 3.f, 0.0f}}, // v12
+    {{-1.0f, -1.0f,  1.0f}, {1.0f,      0.0f}}, // v13
+    {{-1.0f,  1.0f, -1.0f}, {2.f / 3.f, 0.5f}}, // v14
+    {{-1.0f,  1.0f,  1.0f}, {1.0f,      0.5f}}, // v15
+
+    // Vertex data for face 4
+    {{-1.0f, -1.0f, -1.0f}, {1.f / 3.f, 0.0f}}, // v16
+    {{ 1.0f, -1.0f, -1.0f}, {2.f / 3.f, 0.0f}}, // v17
+    {{-1.0f, -1.0f,  1.0f}, {1.f / 3.f, 0.5f}}, // v18
+    {{ 1.0f, -1.0f,  1.0f}, {2.f / 3.f, 0.5f}}, // v19
+
+    // Vertex data for face 5
+    {{-1.0f,  1.0f,  1.0f}, {1.f / 3.f, 0.5f}}, // v20
+    {{ 1.0f,  1.0f,  1.0f}, {2.f / 3.f, 0.5f}}, // v21
+    {{-1.0f,  1.0f, -1.0f}, {1.f / 3.f, 1.0f}}, // v22
+    {{ 1.0f,  1.0f, -1.0f}, {2.f / 3.f, 1.0f}}  // v23
+};
+
+constexpr GLushort indices[] = {
+     0,  1,  2,  3,  3,     // Face 0 - triangle strip ( v0,  v1,  v2,  v3)
+     4,  4,  5,  6,  7,  7, // Face 1 - triangle strip ( v4,  v5,  v6,  v7)
+     8,  8,  9, 10, 11, 11, // Face 2 - triangle strip ( v8,  v9, v10, v11)
+    12, 12, 13, 14, 15, 15, // Face 3 - triangle strip (v12, v13, v14, v15)
+    16, 16, 17, 18, 19, 19, // Face 4 - triangle strip (v16, v17, v18, v19)
+    20, 20, 21, 22, 23      // Face 5 - triangle strip (v20, v21, v22, v23)
+};
+
+// Calculate actual size once
+constexpr size_t vertexCount = std::size(vertices);
+constexpr size_t indexCount = std::size(indices);
+
+
+QPlasmaCubicRender::QPlasmaCubicRender(QWidget* parent, const QImage& cubemap)
+    : QOpenGLWidget(parent), fGLProgram(), fGLTexture(), fGLLabelTexture(), fCubeMap(cubemap),
+      fGLIndexBuf(QOpenGLBuffer::IndexBuffer), fGLArrayBuf(QOpenGLBuffer::VertexBuffer),
+      fGLRotation(0.0f, 0.0f, 0.0f, 1.0f), fMouseFrom(0, 0), fShowFaceLabelsOption()
```

Comment:
GCC emits several `-Wreorder` warnings here
```suggestion
    : QOpenGLWidget(parent), fGLProgram(), fGLArrayBuf(QOpenGLBuffer::VertexBuffer),
      fGLIndexBuf(QOpenGLBuffer::IndexBuffer), fGLTexture(), fGLLabelTexture(),
      fGLRotation(0.0f, 0.0f, 0.0f, 1.0f), fCubeMap(cubemap), fMouseFrom(0, 0),
      fShowFaceLabelsOption()
```